src/dependency/UpdateDependencies.cpp

```diff
@@ -0,0 +1,64 @@
+#include "pch/stdafx.h"
+#include "dependency/UpdateDependencies.h"
+#include "api/ApiException.h"
+
+using UKControllerPlugin::Api::ApiInterface;
+using UKControllerPlugin::Api::ApiException;
+using UKControllerPlugin::Windows::WinApiInterface;
+
+namespace UKControllerPlugin {
+    namespace Dependency {
+
+        void UpdateDependencies(const ApiInterface& api, WinApiInterface& filesystem)
+        {
+            // Download the dependency list and save it to the filesystem
+            LogInfo("Downloading dependency data");
+            nlohmann::json dependencyList;
+            try {
+                dependencyList = api.GetDependencyList();
+                filesystem.WriteToFile("dependencies/dependency-list.json", dependencyList.dump(), true);
+                LogInfo("Downloaded dependency list");
+            } catch (ApiException exception) {
+                LogError("Unable to download dependency list: " + std::string(exception.what()));
+                return;

```

Comment:
Do you need this since it is a void function and signifies the end of execution?

Reply:
Yeah - just forces early termination of the function as we don't wanna try downloading dependencies if we cant download an up-to-date list.